docs/zenith/developer/go/457202-test-build-publish.md

```diff
@@ -0,0 +1,303 @@
+
+1. Add the `node` module as a dependency:
+
+  ```shell
+  dagger mod install github.com/quartz-technology/daggerverse/node@9ce087b83aa8b85f828d7d92ce39bd7c055cfc0f
+  ```
+
+1. Update the generated `main.go` file with the following code:
+
+  ```go
+  package main
+
+  import (
+    "context"
+  )
+
+  type Mymod struct {}
+
+  const defaultNodeVersion = "16"
+
+  func (m *Mymod) buildBase(nodeVersion Optional[string]) *Node {
+    return dag.Node().
+      WithVersion(nodeVersion.GetOr(defaultNodeVersion)).
+      WithNpm().
+      WithSource(dag.Host().Directory(".", HostDirectoryOpts{
+        Exclude: []string{".git", "**/node_modules"},
+      })).
+      Install(nil)
+  }
+  ```
+
+  This function does the following:
+    - It calls the `node` module's `WithVersion()` method via the `dag` client. This method returns a `node` container image with the given Node.js version. This container image is represented as a `Container` object.
+    - It calls the module's `Container.WithNpm()` method, which returns a revised `Container` after adding the `npm` package manager and a cache volume for `npm`.
+    - It calls the module's `Container.WithSource()` method, which returns a revised `Container` including the application source code and a cache volume for Node.js modules.
+    - It calls the module's `Container.Install()` method, which runs `npm install` in the container and returns a revised `Container` including the application's dependencies.
+
+  :::note
+  `dag` is the Dagger client, which is pre-initialized. It contains all the core types (like `Container`, `Directory`, etc.), as well as bindings to any dependencies your module has declared (like `node`).
+  :::
+
+## Step 3: Add a function to test the application
+
+The return value of the `buildBase()` function is a `Container` with the application source code, Node.js runtime and cahe volumes. This is everything needed to test, build and publish the application.
+
+Add a new `Test()` function that runs tests for the example application, by executing the `npm test` command:
+
+```go
+func (m *Mymod) Test(ctx context.Context, nodeVersion Optional[string]) (string, error) {
+  return m.buildBase(nodeVersion).
+    Run([]string{"test", "--", "--watchAll=false"}).
+    Stderr(ctx)
+}
+```
+
+This function does the following:
+
+- It calls the `buildBase()` function to obtain a `Container` with the application source code, Node.js runtime and cache volumes.
+- It calls the module's `Container.Run()` method, which returns a revised `Container` after setting the commands to run in the container image - in this case, the command `npm test -- --watchAll-false`.
+- It uses the `Container.Stderr()` method to return the error stream of the last executed command. No error output implies successful execution (all tests pass).
```

Comment:
You will get output on stderr when tests pass (showing that the tests pass). If tests fail, you should get a non-nil `error` return value here, which is what propagates to the dagger CLI and lets it know something failed.

Reply:
Added